lib/_stream_readable.js

```diff
@@ -267,10 +267,10 @@ function addChunk(stream, state, chunk, addToFront) {
 
 function chunkInvalid(state, chunk) {
   var er;
-  if (!Stream._isUint8Array(chunk) &&
-      typeof chunk !== 'string' &&
+  if (typeof chunk !== 'string' &&
       chunk !== undefined &&
-      !state.objectMode) {
+      !state.objectMode &&
+      !Stream._isUint8Array(chunk)) {
```

Comment:
You are right about this, I will need to check if this does not negatively effect benchmarks.